rust_icu_ecma402/src/listformat.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implements the traits found in [ecma402_traits::listformat].
+
+use ecma402_traits;
+use ecma402_traits::listformat;
+use rust_icu_common as common;
+use rust_icu_ulistformatter as ulfmt;
+use std::fmt;
+
+/// Implements [listformat::Format] using ICU as the underlying implementing library.
+#[derive(Debug)]
+pub struct Format {
+    rep: ulfmt::UListFormatter,
+}
+
+// Full support for styled formatting is available since v67.
+#[cfg(feature = "icu_version_67_plus")]
+pub(crate) mod internal {
+    use rust_icu_sys as usys;
+    use ecma402_traits::listformat::options;
+
+    // Converts the param types from ECMA-402 into ICU.
+    pub fn to_icu_width(style: &options::Style) -> usys::UListFormatterWidth {
```

Comment:
I'm conflicted. If we're making `rust_icu` depend on `ecma402_traits` anyway, maybe we should add standard `From`/`To` conversion traits to `rust_icu`. (This would require some hacks for bindgen code; cf. https://github.com/rust-lang/rust-bindgen/issues/1089.)

Reply:
(This would be in a later CL, of course.)

Reply:
Not sure how automated bindgen would help in this case though, care to elaborate?

But I share your concern -- it reads like an inverted dependency on (fairly low level) rust_icu that it should conform to a set of high level conceptual APIs (ecma 402).  
 
On the other hand, reimplementing Locale in a rust_icu backed implementation of ecma402 seems like we're, well, redoing the work that already exists in rust_icu.